invenio_records_resources/records/systemfields/has_draftcheck.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Persistent identifier status field.
+
+The PIDStatusField is used to check if an associated PID is in a given state.
+For instance:
+
+.. code-block:: python
+
+    class Record():
+        pid = PIDField()
+        is_published = PIDStatusCheckField('pid', status=PIDStatus.REGISTERED)
+
+"""
+
+from invenio_pidstore.models import PIDStatus
+from invenio_records.systemfields import SystemField
+
+from sqlalchemy.orm.exc import NoResultFound
+
+
+class HasDraftCheckField(SystemField):
+    """PID status field which checks against an expected status."""
+
+    def __init__(self, draft_cls):
+        """Initialize the PIDField.
+
+        :param key: Attribute name of the PIDField to use for status check.
+        :param status: The status or list of statuses which will return true.
+        """
+        # assert status, "You must provide a PIDStatus to test against."
+        self._draft_cls = draft_cls
+
+    #
+    # Data descriptor methods (i.e. attribute access)
+    #
+    def __get__(self, record, owner=None):
```

Comment:
For the logic on caching - see https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/systemfields/relatedmodelfield.py#L88-L139